src/service/ratelimit.go

```diff
@@ -75,12 +75,15 @@ func (this *service) reloadConfig(statsManager stats.Manager) {
 		files = append(files, config.RateLimitConfigToLoad{key, snapshot.Get(key)})
 	}
 
-	newConfig := this.configLoader.Load(files, statsManager)
+	this.configLock.Lock()
+	defer this.configLock.Unlock()
```

Comment:
For performance reasons I think you want to keep the locking where it is. I think you can just move the settings construction outside the lock? I'm not sure why it's in there. Seems like that should be static anyway?

Reply:
Agree. Having a second look at it, moving out the constructor should not have any impact.